python/tk_desktop2/websockets/websockets_connection.py

```diff
@@ -186,9 +191,9 @@ def _handle_server_id_request(self, message):
                 "Unexpected error while trying to retrieve the user id: "
                 "No user information was found in this request."
             )
-
-        # validate that the site of the request matches the sg create site
-        if self._origin_site != self._bundle.sgtk.shotgun_url:
+        
+        # validate that the site of the request matches shotgunlocalhost.com or the sg create site
+        if not WebsocketsConnection.localhost_re.match(self._origin_site) and self._origin_site != self._bundle.sgtk.shotgun_url:
```

Comment:
thinking about the support aspect here - when things don't work as expected and debugging is needed - will there be enough debug info emitted to be able to troubleshoot issues? Should we add debug to explain when we decide it's localhost vs external site for example?

Reply:
It's true that I had to inject a remote_pdb call to debug this part. Will add more debug logging.